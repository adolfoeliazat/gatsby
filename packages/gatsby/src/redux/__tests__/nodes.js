const { actions, boundActions } = require(`../actions`)
const nodeReducer = require(`../reducers/nodes`)
const nodeTouchedReducer = require(`../reducers/nodes-touched`)

describe(`Create and update nodes`, () => {
  it(`allows creating nodes`, () => {
    const action = actions.createNode({
      id: `hi`,
      children: [],
      parent: `test`,
      internal: {
        contentDigest: `hasdfljds`,
        mediaType: `test`,
        pluginOwner: `tests`,
        type: `Test`,
      },
      pickle: true,
    })
    expect(action).toMatchSnapshot()
    expect(nodeReducer(undefined, action)).toMatchSnapshot()
  })

  it(`allows updating nodes`, () => {
    const action = actions.createNode({
      id: `hi`,
      children: [],
      parent: `test`,
      internal: {
        contentDigest: `hasdfljds`,
        mediaType: `test`,
        pluginOwner: `tests`,
        type: `Test`,
      },
      pickle: true,
      deep: {
        array: [0, 1, { boom: true }],
      },
    })
    const updateAction = actions.createNode({
      id: `hi`,
      children: [],
      parent: `test`,
      internal: {
        contentDigest: `hasdfljds`,
        mediaType: `test`,
        pluginOwner: `tests`,
        type: `Test`,
      },
      pickle: false,
      deep: {
        array: [1, 2],
      },
      deep2: {
        boom: `foo`,
      },
    })
    let state = nodeReducer(undefined, action)
    state = nodeReducer(state, updateAction)
    expect(state[`hi`].pickle).toEqual(false)
    expect(state[`hi`].deep).toEqual({ array: [1, 2] })
    expect(state[`hi`].deep2).toEqual({ boom: `foo` })
  })

  it(`nodes that are added are also "touched"`, () => {
    const action = actions.createNode({
      id: `hi`,
      children: [],
      parent: `test`,
      internal: {
        contentDigest: `hasdfljds`,
        mediaType: `test`,
        pluginOwner: `tests`,
        type: `Test`,
      },
      pickle: true,
    })
    let state = nodeTouchedReducer(undefined, action)
    expect(state[`hi`]).toBe(true)
  })
})
