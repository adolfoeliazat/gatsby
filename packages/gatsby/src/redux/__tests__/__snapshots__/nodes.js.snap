// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create and update nodes allows creating nodes 1`] = `
Object {
  "payload": Object {
    "children": Array [],
    "id": "hi",
    "internal": Object {
      "contentDigest": "hasdfljds",
      "mediaType": "test",
      "pluginOwner": "tests",
      "type": "Test",
    },
    "parent": "test",
    "pickle": true,
  },
  "plugin": undefined,
  "type": "CREATE_NODE",
}
`;

exports[`Create and update nodes allows creating nodes 2`] = `
Object {
  "hi": Object {
    "children": Array [],
    "id": "hi",
    "internal": Object {
      "contentDigest": "hasdfljds",
      "mediaType": "test",
      "pluginOwner": "tests",
      "type": "Test",
    },
    "parent": "test",
    "pickle": true,
  },
}
`;
